dom/src/fromEvent.ts

```diff
@@ -30,3 +33,38 @@ export function fromEvent(
     } as Producer<Event>,
   );
 }
+
+export function preventDefaultConditional(
+  event: any,
+  preventDefault: PreventDefaultOpt,
+): void {
+  if (preventDefault) {
+    if (typeof preventDefault === 'boolean') {
+      event.preventDefault();
+    } else if (typeof preventDefault === 'function') {
+      if (preventDefault(event)) {
+        event.preventDefault();
+      }
+    } else if (typeof preventDefault === 'object') {
+      const matchObject: (m: {}, o: {}) => boolean = (matcher, obj) =>
```

Comment:
What's the purpose of the matchObject local function if it's only called in one place (just below its declaration)?